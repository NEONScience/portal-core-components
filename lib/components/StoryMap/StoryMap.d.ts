export default WrappedStoryMap;
declare const WrappedStoryMap: any;
