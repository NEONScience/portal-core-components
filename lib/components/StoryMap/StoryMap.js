"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;
var _react = _interopRequireWildcard(require("react"));
var _propTypes = _interopRequireDefault(require("prop-types"));
var _styles = require("@material-ui/core/styles");
var _Link = _interopRequireDefault(require("@material-ui/core/Link"));
var _OpenInNew = _interopRequireDefault(require("@material-ui/icons/OpenInNew"));
var _Theme = _interopRequireDefault(require("../Theme/Theme"));
var _FullWidthVisualization = _interopRequireDefault(require("../FullWidthVisualization/FullWidthVisualization"));
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
function _getRequireWildcardCache(e) { if ("function" != typeof WeakMap) return null; var r = new WeakMap(), t = new WeakMap(); return (_getRequireWildcardCache = function (e) { return e ? t : r; })(e); }
function _interopRequireWildcard(e, r) { if (!r && e && e.__esModule) return e; if (null === e || "object" != typeof e && "function" != typeof e) return { default: e }; var t = _getRequireWildcardCache(r); if (t && t.has(e)) return t.get(e); var n = { __proto__: null }, a = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var u in e) if ("default" !== u && {}.hasOwnProperty.call(e, u)) { var i = a ? Object.getOwnPropertyDescriptor(e, u) : null; i && (i.get || i.set) ? Object.defineProperty(n, u, i) : n[u] = e[u]; } return n.default = e, t && t.set(e, n), n; }
const MIN_IFRAME_WIDTH = 240;
const useStyles = (0, _styles.makeStyles)(theme => ({
  openInNewLink: {
    display: 'block',
    width: '100%',
    textAlign: 'right',
    marginTop: theme.spacing(0.5),
    fontSize: '0.8rem'
  },
  openInNewIcon: {
    fontSize: '0.95rem',
    margin: theme.spacing(0, 0.5, -0.25, 0)
  }
}));
const StoryMap = props => {
  const {
    url,
    title
  } = props;
  const classes = useStyles(_Theme.default);
  const iframeRef = (0, _react.useRef)(null);
  return /*#__PURE__*/_react.default.createElement(_FullWidthVisualization.default, {
    vizRef: iframeRef,
    minWidth: MIN_IFRAME_WIDTH,
    deriveHeightFromWidth: "auto",
    "data-selenium": "story-map-container"
  }, /*#__PURE__*/_react.default.createElement("iframe", {
    src: url,
    title: title || 'Neon Story Map',
    ref: iframeRef,
    frameBorder: "0",
    marginHeight: "0",
    marginWidth: "0",
    style: {
      minWidth: "".concat(MIN_IFRAME_WIDTH, "px"),
      minHeight: "".concat(MIN_IFRAME_WIDTH, "px")
    }
  }), /*#__PURE__*/_react.default.createElement(_Link.default, {
    href: url,
    target: "_blank",
    className: classes.openInNewLink
  }, /*#__PURE__*/_react.default.createElement(_OpenInNew.default, {
    className: classes.openInNewIcon
  }), "Open in New Window"));
};
StoryMap.propTypes = {
  url: _propTypes.default.string.isRequired,
  title: _propTypes.default.string
};
StoryMap.defaultProps = {
  title: null
};
const WrappedStoryMap = _Theme.default.getWrappedComponent(StoryMap);
var _default = exports.default = WrappedStoryMap;