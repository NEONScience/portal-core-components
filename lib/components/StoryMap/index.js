"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "default", {
  enumerable: true,
  get: function () {
    return _StoryMap.default;
  }
});
var _StoryMap = _interopRequireDefault(require("./StoryMap"));
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }