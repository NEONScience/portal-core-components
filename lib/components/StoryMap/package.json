{
  "private": true,
  "name": "story-map",
  "main": "./StoryMap.jsx",
  "module": "./StoryMap.js"
}
