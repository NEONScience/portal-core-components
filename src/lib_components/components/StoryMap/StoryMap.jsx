import React, { useRef } from 'react';
import PropTypes from 'prop-types';

import { makeStyles } from '@mui/styles';
import Link from '@mui/material/Link';
import OpenInNewIcon from '@mui/icons-material/OpenInNew';

import Theme from '../Theme/Theme';
import FullWidthVisualization from '../FullWidthVisualization/FullWidthVisualization';
import { resolveProps } from '../../util/defaultProps';

const MIN_IFRAME_WIDTH = 240;

const useStyles = makeStyles((theme) => ({
  openInNewLink: {
    display: 'block',
    width: '100%',
    textAlign: 'right',
    marginTop: theme.spacing(0.5),
    fontSize: '0.8rem',
  },
  openInNewIcon: {
    fontSize: '0.95rem',
    margin: theme.spacing(0, 0.5, -0.25, 0),
  },
}));

const defaultProps = {
  title: null,
};

const StoryMap = (inProps) => {
  const props = resolveProps(defaultProps, inProps);
  const { url, title } = props;

  const classes = useStyles(Theme);
  const iframeRef = useRef(null);

  return (
    <FullWidthVisualization
      vizRef={iframeRef}
      minWidth={MIN_IFRAME_WIDTH}
      deriveHeightFromWidth="auto"
      data-selenium="story-map-container"
    >
      <iframe
        src={url}
        title={title || 'Neon Story Map'}
        ref={iframeRef}
        frameBorder="0"
        marginHeight="0"
        marginWidth="0"
        style={{ minWidth: `${MIN_IFRAME_WIDTH}px`, minHeight: `${MIN_IFRAME_WIDTH}px` }}
      />
      <Link href={url} target="_blank" className={classes.openInNewLink}>
        <OpenInNewIcon className={classes.openInNewIcon} />
        Open in New Window
      </Link>
    </FullWidthVisualization>
  );
};

StoryMap.propTypes = {
  url: PropTypes.string.isRequired,
  title: PropTypes.string,
};

const WrappedStoryMap = Theme.getWrappedComponent(StoryMap);

export default WrappedStoryMap;
