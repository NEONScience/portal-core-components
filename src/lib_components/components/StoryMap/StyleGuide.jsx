/* eslint-disable react/jsx-one-expression-per-line, jsx-a11y/anchor-is-valid */
import React from 'react';

import Link from '@mui/material/Link';
import Typography from '@mui/material/Typography';

import StoryMap from '@/components/StoryMap/StoryMap';

import DocBlock from '../../../components/DocBlock';
import CodeBlock from '../../../components/CodeBlock';
import ExampleBlock from '../../../components/ExampleBlock';

export default function StyleGuide() {
  const FullWidthVisualizationLink = (
    <Link href="#FullWidthVisualization">FullWidthVisualization</Link>
  );

  return (
    <>

      <DocBlock>
        A component for neatly embedding an ArcGIS Story Map into a page.
      </DocBlock>
      <CodeBlock>
        {`
import StoryMap from 'portal-core-components/lib/components/StoryMap';
        `}
      </CodeBlock>

      <Typography variant="h5" component="h3" gutterBottom>Usage</Typography>

      <DocBlock>
        A StoryMap component requires a <tt>url</tt> prop which should be the URL of the story map
        as hosted elsewhere. It also takes a <tt>title</tt> prop which is not required. If there are
        more than one StoryMap components on the same page it is best to explicitly define a unique
        title for each of them.
      </DocBlock>
      <DocBlock>
        The StoryMap component uses the {FullWidthVisualizationLink} component as a container with
        automatic derived height. This means a StoryMap component will always fill the full width of
        its container and automatically derive its height from the viewport size; mimicking the
        aspect ratio of the viewport at several set break points.
      </DocBlock>

      <ExampleBlock>
        <StoryMap
          url="https://storymaps.arcgis.com/stories/953a6c06a53b44d48c3780867ff692d0"
          title="Central Plains: Getting to Know the NEON Domains"
        />
      </ExampleBlock>
      <CodeBlock>
        {`
<StoryMap
  url="https://storymaps.arcgis.com/stories/953a6c06a53b44d48c3780867ff692d0"
  title="Central Plains: Getting to Know the NEON Domains"
/>
        `}
      </CodeBlock>

    </>
  );
}
