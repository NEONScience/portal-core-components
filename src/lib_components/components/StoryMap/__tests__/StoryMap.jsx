import React from 'react';
import renderer from 'react-test-renderer';

import StoryMap from '../StoryMap';

describe('StoryMap', () => {
  test('Renders correctly with no title', () => {
    const tree = renderer
      .create(<StoryMap url="https://foo.bar" />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  test('Renders correctly with a custom title', () => {
    const tree = renderer
      .create(<StoryMap url="https://foo.bar" title="Foo" />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
