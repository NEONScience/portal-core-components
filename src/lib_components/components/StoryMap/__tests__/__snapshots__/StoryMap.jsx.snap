// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StoryMap Renders correctly with a custom title 1`] = `
<div
  data-selenium="story-map-container"
  style={
    Object {
      "minWidth": "240px",
      "width": "100%",
    }
  }
>
  <iframe
    frameBorder="0"
    marginHeight="0"
    marginWidth="0"
    src="https://foo.bar"
    style={
      Object {
        "minHeight": "240px",
        "minWidth": "240px",
      }
    }
    title="Foo"
  />
  <a
    className="MuiTypography-root MuiLink-root MuiLink-underlineHover makeStyles-openInNewLink-1 MuiTypography-colorPrimary"
    href="https://foo.bar"
    onBlur={[Function]}
    onFocus={[Function]}
    target="_blank"
  >
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root makeStyles-openInNewIcon-2"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M19 19H5V5h7V3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2v-7h-2v7zM14 3v2h3.59l-9.83 9.83 1.41 1.41L19 6.41V10h2V3h-7z"
      />
    </svg>
    Open in New Window
  </a>
</div>
`;

exports[`StoryMap Renders correctly with no title 1`] = `
<div
  data-selenium="story-map-container"
  style={
    Object {
      "minWidth": "240px",
      "width": "100%",
    }
  }
>
  <iframe
    frameBorder="0"
    marginHeight="0"
    marginWidth="0"
    src="https://foo.bar"
    style={
      Object {
        "minHeight": "240px",
        "minWidth": "240px",
      }
    }
    title="Neon Story Map"
  />
  <a
    className="MuiTypography-root MuiLink-root MuiLink-underlineHover makeStyles-openInNewLink-1 MuiTypography-colorPrimary"
    href="https://foo.bar"
    onBlur={[Function]}
    onFocus={[Function]}
    target="_blank"
  >
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root makeStyles-openInNewIcon-2"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M19 19H5V5h7V3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2v-7h-2v7zM14 3v2h3.59l-9.83 9.83 1.41 1.41L19 6.41V10h2V3h-7z"
      />
    </svg>
    Open in New Window
  </a>
</div>
`;
